### ISSUE TITLE
Cannot import flax.training.checkpoints in 0.6.4

### ISSUE THREAD

**Issue by Quasar-Kim** [google/flax#2824]
### System information
- OS Platform and Distribution: Ubuntu 22.04.1 LTS, also in Colab environment
- Flax, jax, jaxlib versions:
  * flax 0.6.4
  * jax 0.3.25
  * jaxlib 0.3.25
- Python version: 3.10.6
- GPU/TPU model and memory: No Accelerator / 16GB

### Problem you have encountered:
With FLAX v0.6.4 I can't import `flax.training.checkpoints` module due to following error:
```
ImportError: cannot import name 'monitoring' from 'jax' (/usr/local/lib/python3.8/dist-packages/jax/__init__.py)
```
This does not happen in v0.6.3.

### What you expected to happen:
The module should be imported.

### Logs, error messages, etc:
Error message from jupyter notebook:
```
ImportError                               Traceback (most recent call last)

[<ipython-input-3-9a234296e658>](https://localhost:8080/#) in <module>
      1 import flax
----> 2 from flax.training import checkpoints

[/usr/local/lib/python3.8/dist-packages/flax/training/checkpoints.py](https://localhost:8080/#) in <module>
     36 from flax import traverse_util
     37 import jax
---> 38 from jax import monitoring
     39 from jax import process_index
     40 from jax import sharding

ImportError: cannot import name 'monitoring' from 'jax' (/usr/local/lib/python3.8/dist-packages/jax/__init__.py)
```

### Steps to reproduce:
[Colab notebook](https://colab.research.google.com/drive/1ZLR1JSJPfaaoTmL7bow8oebqyhhxrqSo?usp=sharing)

**Comment by Quasar-Kim** [comment]
The problem seems to be that `jax.monitoring` is available since `0.4` but flax requires `>=0.3.16`. I was also able to resolve this problem by upgrading `jax` and `jaxlib` to `0.4.2`.

**Comment by 1kaiser** [comment]
!python -m pip install jax==0.4.2 jaxlib==0.4.2
